irctest/server_tests/test_lusers.py

```diff
@@ -118,10 +121,24 @@ class LusersUnregisteredTestCase(LusersTestCase):
     def testLusers(self):
         self.doLusersTest()
 
+    def _synchronize(self, client_name):
```

Comment:
Can we incorporate this directly into the standard `getMessages`? i.e., accept either the correct PONG parameter or `ERR_NOTREGISTERED` as the synchronizing event?

Ideally this would not sleep either, although it appears that `getMessages` currently has a sleep.

Reply:
I'd rather not, because it may be returned because of an other command the test sent to the server, so it would be very confusing.

Reply:
Hmm. The client should transition from unregistered to registered, exactly once, and never return. So we could detect `001` and only accept `ERR_NOTREGISTERED` before `001` has been seen?

Reply:
it doesn't address the issue. I'm thinking of a test sending `PRIVMSG` before registering, for example.

Reply:
That issue exists with `_synchronize` too, right? The suggestion is just that we expose it as little as possible?

How about as a `preregistration` boolean kwarg to `getMessages`?

Reply:
> That issue exists with _synchronize too, right? The suggestion is just that we expose it as little as possible?

Yes, exactly

> How about as a preregistration boolean kwarg to getMessages?

Hmm... maybe. Let's keep it like this, and add that kwarg later if the need arises?

Reply:
sgtm